pxtblocks/code-validation/validateBlocksExist.ts

```diff
@@ -0,0 +1,40 @@
+
+namespace pxt.blocks {
+    export function validateBlocksExist({ usedBlocks, requiredBlockCounts }: {
+        usedBlocks: Blockly.Block[],
+        requiredBlockCounts: pxt.Map<number>,
+    }): {
+        missingBlocks: string[],
+        disabledBlocks: string[],
+        insufficientBlocks: string[],
+    } {
+        let missingBlocks: string[] = [];
+        let disabledBlocks: string[] = [];
+        let insufficientBlocks: string[] = [];
+        const userBlocksEnabledByType = usedBlocks?.reduce((acc: pxt.Map<number>, block) => {
+            acc[block.type] = (acc[block.type] || 0) + (block.isEnabled() ? 1 : 0);
+            return acc;
+        }, {});
+
+        for (const [requiredBlockId, requiredCount] of Object.entries(requiredBlockCounts)) {
+            const countForBlock = userBlocksEnabledByType[requiredBlockId];
+            if (countForBlock === undefined) {
+                // user did not use a specific block
+                missingBlocks.push(requiredBlockId);
+            } else if (!countForBlock) {
+                // all instances of block are disabled
+                disabledBlocks.push(requiredBlockId);
+            } else if (countForBlock < requiredCount) {
```

Comment:
Added this branch, though not yet used; to allow us to check if a block is used enough when the same block is used multiple times